### PR TITLE
feat(server): add ServiceAccount info to api/v1/userinfo and ui user tab

### PR DESCRIPTION
**Pull request by tczhao**
Resolves: #4936

Signed-off-by: Tianchu Zhao <evantczhao@gmail.com>

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

File: pkg/apiclient/info/info.proto [hunk 1]
```diff
@@ -26,6 +26,7 @@ message GetUserInfoResponse {
     repeated string groups = 3;
     string email = 4;
     bool emailVerified = 5;
+    string serviceAccount = 6;
```
Comment (alexec) [inline pkg/apiclient/info/info.proto:29]: I think it should probably be consistent with PodSpec which would be serviceAccountName:

https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/

**Review by alexec — COMMENTED** [review]
Needs @jessesuen approval for proto changes.

**Review by jessesuen — APPROVED** [review]
Agree with @alexec. Conditional approval provided we rename to serviceAccountName.

**Review by alexec — APPROVED** [review]

**Comment by alexec** [comment]
I think e2e test failure is unrelated. Re-running.

**Comment by alexec** [comment]
Please sync with master to fix conflicts introduced by v2.

**Review by alexec — CHANGES_REQUESTED** [review]
Needs rebase/merge from master.

**Comment by tczhao** [comment]
> Needs rebase/merge from master.

Done

**Review by alexec — APPROVED** [review]

**Comment by alexec** [comment]
@simster7 we might want to back-part this one.

**Pull request merged**